Magento2/Sniffs/Annotation/MethodArgumentsSniff.php

```diff
@@ -617,6 +622,104 @@ public function process(File $phpcsFile, $stackPtr)
         );
     }
 
+    /**
+     * Check method if it has defined return & parameter types, then it doesn't need DocBlock
+     *
+     * @param File $phpcsFile
+     * @param int $stackPtr
+     * @return bool
+     */
+    private function checkIfMethodNeedsDocBlock(File $phpcsFile, $stackPtr) : bool
+    {
+        if ($this->checkIfNamespaceContainsApi($phpcsFile)) {
+            // Note: API classes MUST have DocBlock because it uses them instead of reflection for types
+            return true;
+        }
+        $foundAllParameterTypes = true;
+        $methodParameters = $phpcsFile->getMethodParameters($stackPtr);
+        foreach ($methodParameters as $parameter) {
+            if (!$parameter['type_hint']) {
+                return true; // doesn't have type hint
+            }
+        }
+        $methodProperties = $phpcsFile->getMethodProperties($stackPtr);
+        $foundReturnType = !!$methodProperties['return_type'];
+        if (!$foundReturnType) {
+            $methodName = $phpcsFile->getDeclarationName($stackPtr);
+            // Note: __construct and __destruct can't have return types
+            if ('__construct' !== $methodName && '__destruct' !== $methodName) {
+                return true;
+            }
+        }
+        $complexity = $this->getMethodComplexity($phpcsFile, $stackPtr);
+        return $complexity > static::MAXIMUM_COMPLEXITY_ALLOWED_FOR_NO_DOCBLOCK;
+    }
+
+    /**
+     * Check if namespace contains API
+     *
+     * @param File $phpcsFile
+     * @return bool
+     */
+    private function checkIfNamespaceContainsApi(File $phpcsFile) : bool
+    {
+        $namespaceStackPtr = $phpcsFile->findNext(T_NAMESPACE, 0);
+        if (!is_int($namespaceStackPtr)) {
+            return false;
+        }
+        $tokens = $phpcsFile->getTokens();
+        // phpcs:ignore Squiz.ControlStructures.ForLoopDeclaration.SpacingAfterOpen,Generic.CodeAnalysis.ForLoopWithTestFunctionCall.NotAllowed
+        for (
+            // phpcs:ignore Squiz.ControlStructures.ForLoopDeclaration.SpacingAfterFirst
+            $index = $namespaceStackPtr;
+            // phpcs:ignore Squiz.ControlStructures.ForLoopDeclaration.SpacingAfterSecond
```

Comment:
Why not respect the coding standard?
```suggestion
```

Reply:
I've resolved these in 17e120f by replacing array_key_exists with the null coalescing operator.